cristin-commons/src/main/java/no/unit/nva/cristin/model/CristinQuery.java

```diff
@@ -36,43 +23,42 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
-import no.unit.nva.cristin.projects.model.nva.ProjectStatus;
 import nva.commons.core.paths.UriWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @SuppressWarnings({"Unused", "LooseCoupling"})
-public class CristinQuery {
+public abstract class CristinQuery<T extends Enum<T> & IParameterKey> {
 
-    protected final transient Map<QueryParameterKey, String> pathParameters;
-    protected final transient Map<QueryParameterKey, String> queryParameters;
-    protected final transient Set<QueryParameterKey> otherRequiredKeys;
-    protected transient boolean isNvaQuery = true;
+    protected static final Logger logger = LoggerFactory.getLogger(CristinQuery.class);
+    protected final transient Map<T, String> pathParameters;
+    protected final transient Map<T, String> queryParameters;
+    protected final transient Set<T> otherRequiredKeys;
 
     protected CristinQuery() {
         queryParameters = new ConcurrentHashMap<>();
         pathParameters = new ConcurrentHashMap<>();
         otherRequiredKeys = new HashSet<>();
     }
 
-    public static CristinQueryBuilder builder() {
-        return new CristinQueryBuilder();
+
+    public static String getUnitIdFromOrganization(String organizationId) {
```

Comment:
is organizationId an url or a number of format 1234.12.12.12?

Reply:
it returns extractLastPathElement,
it is validated with `ORGANIZATION_IDENTIFIER_PATTERN = "^(?:[0-9]+\\.){3}[0-9]{1,3}$";`